tests/test_security_curve.cpp

```diff
@@ -40,6 +40,10 @@
 #  include <unistd.h>
 #endif
 
+#include "../src/tweetnacl.h"
```

Comment:
But the code is leaking - personally I would much, much rather prefer copying part of the implementation in the test case client. Having to jump through the hoops to include source code directly in the test binary is a bit messy and I'm afraid it will cause problems in funkier build systems like visual studio :-/ I can take care of that in a while.